website/docs/r/ruleset.html.markdown

```diff
@@ -121,6 +148,15 @@ The following arguments are supported:
 * `uri` - (Optional) List of URI properties to configure for the ruleset rule when performing URL rewrite transformations (refer to the [nested schema](#nestedblock--action-parameters-uri)).
 * `version` - (Optional)
 
+<a id="nestedblock--ratelimiting-parameters"></a>
+**Nested schema for `ratelimit`**
+
+* `characteristics` - (Optional) List of parameters that define how Cloudflare tracks the request rate for this rule.
```

Comment:
Quick note about the optional/mandatory fields (sorry for only noticing this now).

For rate limiting rules, only "mitigation_expression" is optional. However, all the fields are marked as optional.

It would probably be more useful to say if the fields are mandatory/optional in the context of the current nested schema. However, all other sections consider the global rule definition context, so they would also have to be changed.

Reply:
The value will come from the Terraform schema which unfortunately doesn't have the same level of context as the API itself so in that aspect, and what we report here, the fields are optional. In the past, we've leant on adequate documentation in the API or developer documentation to provide more insight into available configurations.